camelot/view/action_steps/application.py

```diff
@@ -75,16 +77,18 @@ def render( self, gui_context ):
                 window = widget
                 break
 
-        main_window = MainWindow( gui_context=main_window_context, parent=None, window=window )
+        main_window_proxy = MainWindowProxy( gui_context=main_window_context, parent=None, window=window )
+        register( main_window_proxy, main_window_proxy )
+
         gui_context.workspace = main_window_context.workspace
-        main_window.window.setWindowTitle( self.window_title )
-        return main_window
+        main_window_proxy.parent().setWindowTitle( self.window_title )
+        return main_window_proxy
```

Comment:
ik denk dat de render method best gewoon de mainwindow widget kan blijven teruggeven,
dat is consistent met de andere render methods, en het gebruik ervan in de unit tests

Reply:
Dat is inderdaad beter, heb het al aangepast.